geode-core/src/main/java/org/apache/geode/cache/query/internal/index/CompactRangeIndex.java

```diff
@@ -1506,7 +1506,8 @@ private void applyProjectionForIndexInit(List currentRuntimeIters)
           : modifiedIndexExpr.evaluate(this.initContext);
 
       if (indexKey == null) {
-        indexKey = IndexManager.NULL;
+        // indexKey = IndexManager.NULL;
```

Comment:
I don't think we can do this without losing some capabilities... such as != queries.  This line of code also affects regular ranged indexes and not just map indexes.  

I think we'd have to find some way for the index to supply the correct entries for those types of queries.  Or find a better way to store the entries in the index.

Reply:
I think you are right. There is trouble with the != queries.
Shouldn't they not use the index?
According to the documentation, indexes are not used in expressions that contain NOT(https://geode.apache.org/docs/guide/19/developing/query_index/indexing_guidelines.html) . I think the same could be done for queries using != which is very similar to using NOT.
Nevertheless, when I checked the code, it seems that queries with NOT also used the indexes.